src/JustSaying/AwsTools/MessageHandling/IMessageCoordinator.cs

```diff
@@ -0,0 +1,14 @@
+using System.Threading;
+using System.Threading.Tasks;
+using JustSaying.Messaging.MessageProcessingStrategies;
+
+namespace JustSaying.AwsTools.MessageHandling
+{
+    public interface IMessageCoordinator
+    {
+        Task ListenAsync(CancellationToken cancellationToken);
+        void WithMessageProcessingStrategy(IMessageProcessingStrategy messageProcessingStrategy);
```

Comment:
As this is an interface, I'd probably go with a property that can be null. "Builder" type method patterns don't work well with interfaces.